src/Symfony/Component/Notifier/Bridge/AmazonSns/AmazonSnsTransportFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\AmazonSns;
+
+use AsyncAws\Sns\SnsClient;
+use Symfony\Component\Notifier\Exception\UnsupportedSchemeException;
+use Symfony\Component\Notifier\Transport\AbstractTransportFactory;
+use Symfony\Component\Notifier\Transport\Dsn;
+use Symfony\Component\Notifier\Transport\TransportInterface;
+
+/**
+ * @author Adrien Chinour <github@chinour.fr>
+ */
+final class AmazonSnsTransportFactory extends AbstractTransportFactory
+{
+    public function create(Dsn $dsn): TransportInterface
+    {
+        $scheme = $dsn->getScheme();
+
+        if ('sns' !== $scheme) {
+            throw new UnsupportedSchemeException($dsn, 'sns', $this->getSupportedSchemes());
+        }
+
+        $host = 'default' === $dsn->getHost() ? null : $dsn->getHost();
+        $port = $dsn->getPort();
+
+        $options = [
+                'region' => $dsn->getOption('region') ?: 'eu-west-1',
+                'profile' => $dsn->getOption('profile'),
+                'accessKeyId' => $dsn->getUser(),
+                'accessKeySecret' => $dsn->getPassword(),
```

Comment:
Lets only add these if they have a value. 

Ie, there is no need to have an empty `accessKeyId` or to redefine `region`. So if `$dsn->getUser()` has a specified value, then add it to `$options`